servant-foreign/src/Servant/Foreign/Internal.hs

```diff
@@ -120,142 +122,175 @@ type family Elem (a :: *) (ls::[*]) :: Constraint where
   Elem a (a ': list) = ()
   Elem a (b ': list) = Elem a list
 
-class HasForeign (layout :: *) where
+class HasForeignType lang a where
```

Comment:
Could you add a short haddock description (in particular mentioning how dynamically typed languages should deal with this)?